editor/utils/heartbeat.js

```diff
@@ -0,0 +1,153 @@
+import store from '../store';
+import {
+	getEditedPostTitle,
+	getEditedPostExcerpt,
+	getEditedPostContent,
+	getCurrentPostId,
+	isPostAutosaveDirty,
+} from '../store/selectors';
+import {
+	toggleAutosave,
+	removeNotice,
+	toggleNetworkIsConnected,
+	showDisconnectionNotice,
+	resetAutosave,
+	updateAutosaveStatusMessage,
+} from '../store/actions';
+
+/**
+ * WordPress dependencies
+ */
+import { doAction } from '@wordpress/hooks';
+
+const DISCONNECTION_NOTICE_ID = 'DISCONNECTION_NOTICE_ID';
+
+/**
+ * Set up the heartbeat functionality for Gutenberg.
+ */
+export function setupHeartbeat() {
+	const $document = jQuery( document );
+
+	/**
+	 * Configure heartbeat to refresh the wp-api nonce, keeping the editor authorization intact.
+	 *
+	 * @todo update the _wpnonce used by autosaves.
+	 */
+	$document.on( 'heartbeat-tick', ( event, response ) => {
```

Comment:
Should the now-duplicated logic for `heartbeat-tick` in `edit-post/index.js` be removed?

Reply:
yes, this was lost in a previous merge

Reply:
removed in https://github.com/WordPress/gutenberg/pull/4218/commits/5459e20fe5715ffab1bf0e3393045f2eefc1737c